internal/controllers/subnet/actuator.go

```diff
@@ -263,12 +265,78 @@ func (actuator subnetActuator) DeleteResource(ctx context.Context, obj orcObject
 	return progress.WrapError(actuator.osClient.DeleteSubnet(ctx, osResource.ID))
 }
 
+func (actuator subnetActuator) updateResource(ctx context.Context, obj orcObjectPT, osResource *osResourceT) progress.ReconcileStatus {
+	log := ctrl.LoggerFrom(ctx)
+	resource := obj.Spec.Resource
+	if resource == nil {
+		// Should have been caught by API validation
+		return progress.WrapError(
+			orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, "Update requested, but spec.resource is not set"))
+	}
+
+	var updateOpts subnets.UpdateOpts
+
+	name := getResourceName(obj)
+	if osResource.Name != name {
+		updateOpts.Name = &name
+	}
+
+	description := string(ptr.Deref(resource.Description, ""))
+	if osResource.Description != description {
+		updateOpts.Description = &description
+	}
+
+	missingAllocationPool := false
+	allocationPools := make([]subnets.AllocationPool, len(resource.AllocationPools))
+	for i := range resource.AllocationPools {
+		allocationPools[i].Start = string(resource.AllocationPools[i].Start)
+		allocationPools[i].End = string(resource.AllocationPools[i].End)
+
+		found := false
+		for _, pool := range osResource.AllocationPools {
+			if pool.Start == allocationPools[i].Start &&
+				pool.End == allocationPools[i].End {
+				found = true
+				break
+			}
+		}
+		if !found {
+			missingAllocationPool = true
+		}
+	}
+	if missingAllocationPool || len(resource.AllocationPools) != len(osResource.AllocationPools) {
+		updateOpts.AllocationPools = allocationPools
+	}
+
+	if reflect.ValueOf(updateOpts).IsZero() {
+		log.V(logging.Debug).Info("No changes")
+		return nil
+	}
+
+	updateOpts.RevisionNumber = &osResource.RevisionNumber
+
+	osResource, err := actuator.osClient.UpdateSubnet(ctx, osResource.ID, updateOpts)
+
+	// We should require the spec to be updated before retrying an update which returned a conflict
+	if orcerrors.IsConflict(err) {
+		err = orcerrors.Terminal(orcv1alpha1.ConditionReasonInvalidConfiguration, "invalid configuration updating resource: "+err.Error(), err)
+	}
+
+	if err != nil {
+		return progress.WrapError(err)
+	}
+
+	// FIXME(mandre) We would not need schedule a new reconcile if we returned the osResource directly
+	return progress.NeedsRefresh()
```

Comment:
Also, this causes the object to be stuck with the `Resource status will be refreshed` progressing condition (reason for some of the tests failing).

Reply:
Found my issue, I tried to update the allocation pool to an empty value, while we get a default one when it's not set in the spec.